daft/expressions/expressions.py

```diff
@@ -1076,16 +1076,27 @@ def __repr__(self) -> str:
 class ExpressionImageNamespace(ExpressionNamespace):
     """Expression operations for image columns."""
 
-    def decode(self) -> Expression:
+    def decode(self, on_error: Literal["raise"] | Literal["null"] = "raise") -> Expression:
```

Comment:
If we expect this to always have boolean semantics, a better user-facing API might be `raise_on_error: bool` or the like.

Reply:
I was following the semantics of our current `.url.download(on_error: "null" | "raise")`, so that our APIs could be more consistent.